### ISSUE TITLE
Auto token renew is not working.

### ISSUE THREAD

**Issue by pbollaram** [okta/okta-auth-js#892]
first, I thought it would be an easy setup. but for some reason, auto-renewal of tokens is not happening.

I have followed the below example
https://github.com/okta/samples-js-react/tree/master/custom-login

here is my setup.
```
oidc: {
    clientId: CLIENT_ID,
    issuer: ISSUER,
    redirectUri: REDIRECT_URI,
    scopes: ['openid', 'profile', 'email'],
    pkce: true,
    disableHttpsCheck: OKTA_TESTING_DISABLEHTTPSCHECK,
    useInteractionCode: USE_INTERACTION_CODE,
    tokenManager:{
      autoRenew: true,
      secure: true,
      storage: 'localStorage',
    }
  }
```
  then
  const oktaAuth = new OktaAuth(config.oidc);
  and also included Security tag
  
  <Security
        oktaAuth={oktaAuth}
        onAuthRequired={customAuthHandler}
        restoreOriginalUri={restoreOriginalUri}
      >
   </Security>
  
  but when it is time for token renewal, I was just flashed with the login widget and the App never sent a network request to /token endpoint
 
I have enabled refresh token grant type on app settings too. I am just confused about where I was making the mistake
is it because i am using custom widget?
  
  
![image](https://user-images.githubusercontent.com/65185492/128284777-b58f6669-68cb-4a2d-bfbe-177b5153ef75.png)

**Comment by pbollaram** [comment]
i see following message and network call when the token gets expired>

<oktadomain>/**oauth2/v1/authorize**?client_id=<id>&code_challenge=ED6SjJD575ravPD29TI-wANMIgOpRFfuUonPMmb2GQU&code_challenge_method=S256&nonce=IKBRk9dDnqO5XItQR0sa7SQjJEJ4lTF3IK2wZa3s0gVUyoo0SQcq2bfes4slUTws&prompt=none&redirect_uri=https%3A%2F%2Fqm-dev.gcs.paloaltonetworks.local%2Fimplicit%2Fcallback&response_mode=okta_post_message&response_type=code&state=xgLcMzzQcM6wLkHXcWnhgv0VBDuTCsvnX4Ox4St05KML99Xiw2H2xvGoTAFA5oU2&scope=openid%20profile%20email:42 

**error.errorCode: login_required, error.description: The client specified not to prompt, but the user is not logged in.**

**Comment by oleksandrpravosudko-okta** [comment]
Thanks for reaching out @pbollaram.

Please note that refresh token grant requires [`offline_access`](https://developer.okta.com/docs/guides/refresh-tokens/get-refresh-token/#get-a-refresh-token-with-the-code-flow) scope to be specified.

As for `The client specified not to prompt, but the user is not logged in.` error - it is likely to happen when 3p cookies are blocked or token lifetime exceeds Okta session duration.

**Comment by pbollaram** [comment]
thanks for the quick reply @oleksandrpravosudko-okta 
adding offline_access to scope leading to 400 error on **/oauth2/v1/token** endpoint when signing in.|

The added scope in config js
 scopes: ['openid', 'profile', 'email','offline_access'],

{error: "invalid_scope"
error_description: "Browser requests to the token endpoint may not include the offline_access scope."}

**Comment by pbollaram** [comment]
I understand that about the error - The client specified not to prompt, but the user is not logged in because as per settings, okta is waiting for 30sec (the previous token is valid for 30sec). my only concern is why it is not even trying to refresh my token.

**Comment by aarongranick-okta** [comment]
@pbollaram 

- The error "The client specified not to prompt" which is coming from a call to `/authorize` (in an iframe) is expected when trying to renew tokens (without offline_access) and the session is either closed/expired or 3rd party cookies are being blocked.  It basically means the session cookie is invalid.

- The error "Browser requests to the token endpoint.." from the call to `/token` is unexpected, but does show that the client is trying to use offline_access / refresh tokens.

From the URLs you posted it seems you like you are using the "Org" authorization server for the issuer. https://github.com/okta/okta-auth-js#about-the-issuer

Generally, custom scoped access tokens require using a "custom" authorization server. You should have a default one created at `https://<your okta org>/oauth2/default`. Try using this for your issuer - hopefully this corrects this issue!

https://developer.okta.com/docs/concepts/auth-servers/#which-authorization-server-should-you-use

**Comment by pbollaram** [comment]
I have updated my ISSUER from https://<your okta org> to https://<your okta org>//oauth2/default. but now I am unable to log in at all.

Access to fetch at 'https://<your okta org>/oauth2/default/.well-known/openid-configuration' from origin 'http://localhost:3000' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
fetchRequest.ts:51 GET https://<your okta org>/oauth2/default/.well-known/openid-configuration net::ERR_FAILED.

if I remove /oauth2/default then I am able to log in but the refresh token is not happening.

**Comment by vijetmahabaleshwar-okta** [comment]
@pbollaram 
Can you check if http://localhost:3000 is added to your list of trusted origins?
In your okta org go to Security -> API -> Trusted Origins 
If not, add it as shown in the screenshot below  

<img width="1188" alt="Screen Shot 2021-08-05 at 4 54 46 PM" src="https://user-images.githubusercontent.com/27521424/128435856-80e834b7-0e4f-4215-8555-fd50587f27d2.png">

**Comment by pbollaram** [comment]
@vijetmahabaleshwar-okta 
yes it is. that is why I am able to login if I configure my ISSUER as https://{org}

the above CORS error was shown when I make changes to ISSUER as suggested by @aarongranick-okta.
the change suggested was to use /oauth2/default

my company rep suggested me using ISSUER as https://{org} so not sure how this authorization default server was set up.

**Comment by oleksandrpravosudko-okta** [comment]
@pbollaram can you please confirm 'default' authorization server URI exists(has the `/default` section in the Issuer URI column) in your org? 
<img width="587" alt="Screenshot 2021-08-02 at 16 20 18" src="https://user-images.githubusercontent.com/71440851/128478507-46147b01-3118-49bc-923b-cf61e9cad29e.png">

It might not in 'preview' orgs - in such case you will need to specify 'default' authorization server id.
<img width="764" alt="Screenshot 2021-08-06 at 11 12 19" src="https://user-images.githubusercontent.com/71440851/128479388-a552ad0a-05d5-4266-b028-6073be0a83a6.png">

**Comment by pbollaram** [comment]
okay here is my finding.
I did copy the below URL to the browser and updated my client to get a code from the authorization server( includes offline_access scope).

![image](https://user-images.githubusercontent.com/65185492/128542481-8e8059a2-27cc-4312-a673-5281a080aa71.png)

Then I copied the code from response and did a curl to see the response

![image](https://user-images.githubusercontent.com/65185492/128543034-ae11ba44-3c2e-4fe9-a5d6-7244bdfdbf39.png)

and response include td_token, access_token 

and this is not happening with okta SDK or OKTA widget that i am using.

**Comment by pbollaram** [comment]
I have unselected the Refresh token from settings and token refresh started working for me.
